src/components/video-texture-target.js

```diff
@@ -1,18 +1,86 @@
 import { disposeTexture } from "../utils/material-utils";
 import { createVideoOrAudioEl } from "../utils/media-utils";
+import { findNode } from "../utils/three-utils";
+
+/**
+ * @component video-texture-source
+ * This component is intended to be used on entities with a Camera Object3D as a child.
+ * That camera is used to render the scene to a WebGLRenderTarget of the specified resolution
+ * at a maximum of the specified frame rate. It will only render a frame if something sets
+ * it's textureNeedsUpdate property to true. Currently video-texture-target does this
+ * whenever its material is used during the primary camera render (as in, its in view).
+ */
+AFRAME.registerComponent("video-texture-source", {
+  schema: {
+    resolution: { type: "vec2", default: [1280, 720] },
+    fps: { default: 15 }
+  },
+
+  init() {
+    this.camera = findNode(this.el.object3D, n => n.isCamera);
+    this.camera.aspect = this.data.resolution[0] / this.data.resolution[1];
+
+    // TODO currently if a video-texture-source tries to render itself it will fail with a warning.
+    // If we want to support this we will need 2 render targets to swap back and forth.
+    this.renderTarget = new THREE.WebGLRenderTarget(this.data.resolution[0], this.data.resolution[1], {
+      format: THREE.RGBAFormat,
+      minFilter: THREE.LinearFilter,
+      magFilter: THREE.NearestFilter,
+      encoding: THREE.GammaEncoding,
+      depth: false,
+      stencil: false
+    });
+
+    const texture = this.renderTarget.texture;
+    texture.matrixAutoUpdate = false;
+    texture.matrix.scale(1, -1);
+    texture.matrix.translate(0, 1);
+
+    this.textureNeedsUpdate = false;
+  },
+
+  tock(time) {
+    if (!this.textureNeedsUpdate) return;
+
+    if (time < this.lastRenderTime + 1000 / this.data.fps) return;
+
+    // TODO handle hiding UI and showing first person head
+    // Once thats done also rework camrea-tool to use this to do its actual rendering
+
+    const sceneEl = this.el.sceneEl;
+    const renderer = this.renderer || sceneEl.renderer;
+
+    const tmpVRFlag = renderer.vr.enabled;
+    const tmpOnAfterRender = sceneEl.object3D.onAfterRender;
+    delete sceneEl.object3D.onAfterRender;
+    renderer.vr.enabled = false;
+
+    renderer.setRenderTarget(this.renderTarget);
+    renderer.render(sceneEl.object3D, this.camera);
```

Comment:
In order to create a mirror, do you invert the camera's x axis? (scale.x = -scale.x)?

Reply:
I think that's right yeah. The user has full control over the camera properties they put into the gltf so this should already be possible if they want to go for that effect.